Assets/Scripts/Player/HaltPlayer.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HaltPlayer : MonoBehaviour {
+	[SerializeField] private BoxCollider2D hasMetTrigger;
+	[SerializeField] private VignetteHighlight vignetteHighlightPrefab;
+
+	private VignetteHighlight highlight;
+
+	private PlayerController playerController;
+	private bool hasMetOnce = false;
+
+	private void Start() {
+		playerController = GameObject.FindWithTag("Player").GetComponentUnlessNull<PlayerController>();
+		hasMetOnce = false;
+		hasMetTrigger.enabled = false;
+	}
+
+	private void OnTriggerEnter2D(Collider2D other) {
+		if (other.attachedRigidbody.CompareTag("Player") && !other.isTrigger && !hasMetOnce)
```

Comment:
This is definetly overkill, but reordering the comparisons like this is more performant.
Checking booleans is cheap so they are done first to allow an early out which avoids unecessary calls to `CompareTag`. Compare tag is slower because it compares strings which is done character by character.
```suggestion
		if (!hasMetOnce && !other.isTrigger && other.attachedRigidbody.CompareTag("Player"))
```